File: web/src/services/home.service.ts
import apiClient from '../config/api';
import type { HomeData, DashboardData } from '../types/property';

export const homeService = {
  async getHomeData(featuredLimit = 10, newLimit = 10): Promise<HomeData> {
    const response = await apiClient.get('/home', {
      params: { featuredLimit, newLimit },
    });
    return response.data;
  },

  async getDashboardData(): Promise<DashboardData> {
    const response = await apiClient.get('/home/dashboard');
    return response.data;
  },
};

File: mobile/lib/services/home_service.dart
import '../config/api_client.dart';
import '../models/property.dart';

class HomeService {
  final _apiClient = ApiClient();

  Future<HomeData> getHomeData({int featuredLimit = 10, int newLimit = 10}) async {
    try {
      final response = await _apiClient.dio.get(
        '/home',
        queryParameters: {
          'featuredLimit': featuredLimit,
          'newLimit': newLimit,
        },
      );
      return HomeData.fromJson(response.data);
    } catch (e) {
      rethrow;
    }
  }

  Future<DashboardData> getDashboardData() async {
    try {
      // Token will be added automatically by the interceptor
      final response = await _apiClient.dio.get('/home/dashboard');
      return DashboardData.fromJson(response.data);
    } catch (e) {
      rethrow;
    }
  }
}
